packages/strapi-admin/admin/src/containers/MarketplacePage/index.js

```diff
@@ -10,20 +10,23 @@ import Wrapper from './Wrapper';
 
 const MarketPlacePage = () => {
   const history = useHistory();
-  const { autoReload, currentEnvironment, formatMessage, plugins } = useGlobalContext();
+  const { autoReload, emitEvent, currentEnvironment, formatMessage, plugins } = useGlobalContext();
   const { error, isLoading, data } = useFetchPluginsFromMarketPlace();
 
   if (isLoading || error) {
     return <LoadingIndicatorPage />;
   }
 
   const handleDownloadPlugin = async pluginId => {
+    emitEvent('willInstallPlugin', { pluginId });
```

Comment:
For other events with use the property plugin for the name of the plugin we should probably make this consistent

Reply:
ok, did you mean `{ plugin: pluginId }`.  I made that update but if I misunderstood just let me know 🙃